flang/include/flang/Semantics/tools.h

```diff
@@ -212,6 +212,16 @@ inline bool IsCUDADeviceContext(const Scope *scope) {
   return false;
 }
 
+inline bool HasCUDAAttr(const Symbol &sym) {
+  if (const auto *details =
+          sym.GetUltimate().detailsIf<semantics::ObjectEntityDetails>()) {
```

Comment:
```suggestion
  if (const auto *details{
          sym.GetUltimate().detailsIf<semantics::ObjectEntityDetails>()}) {
```

Suggestion probably needs formatting though.

Reply:
Thanks! I always miss the brace init switching between different code base rule!